### PR TITLE
Make synchronous query snippets testable

### PR DESCRIPTION
**Pull request by tseaver**
~~Uses #1833 as a base.~~

### CODE REVIEW

**Comment by tseaver** [comment]
Rebased after merging #1833 to `master`.

File: docs/bigquery-usage.rst [hunk 1]
```diff
@@ -25,6 +25,7 @@ Authentication / Configuration
      >>> from gcloud import bigquery
      >>> client = bigquery.Client()
 
+
```
Comment (daspecster) [inline docs/bigquery-usage.rst]: Are these extra new lines needed for something?

**Comment by daspecster** [comment]
Can we squash this before merging?

File: docs/bigquery-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/bigquery-usage.rst]: They don't affect generated output:  they just delineate sections visually (a bit like PEP8 for top-level module `def` and `class` statements.

File: docs/bigquery_snippets.py [hunk 2]
```diff
@@ -68,10 +68,10 @@ def delete(self):
 
 
 @snippet
-def client_list_datasets(client, to_delete):  # pylint: disable=unused-argument
+def client_list_datasets(client, _):
```
Comment (daspecster) [inline docs/bigquery_snippets.py]: I don't know why I didn't think about this before! 👍

**Comment by daspecster** [comment]
LGTM, I would still squash but whatever you think.

**Comment by tseaver** [comment]
@daspecster I rebased away the first two net-noop commits.  I will merge when Travis passes.

**Pull request merged**